popmon/visualization/utils.py

```diff
@@ -46,6 +47,128 @@ def xtick(lab, top):
     return lab
 
 
+def convert_time_delta(nanoseconds):
+    td = datetime.timedelta(seconds=nanoseconds // 1e9)
+    time_delta = ""
+    if td.days // 7 > 0:
+        time_delta = str(td.days // 7) + "w"
+
+    if td.days % 7 > 0:
+        time_delta = time_delta + " " + str(td.days % 7) + "d"
+
+    if td.seconds // 3600 > 0:
+        time_delta = time_delta + " " + str(td.seconds // 3600) + "h"
+
+    if (td.seconds // 60) % 60 > 0:
+        time_delta = time_delta + " " + str((td.seconds // 60) % 60) + "m"
+
+    if td.seconds % 60 > 0:
+        time_delta = time_delta + " " + str(td.seconds % 60) + "s"
+
+    return time_delta
+
+
+def get_summary_table(
+    num_features,
+    time_bins,
+    time_axis,
+    reference_type,
+    time_width,
+    offset,
+    max_timestamp,
+):
+
+    bin_width = convert_time_delta(time_width)
+    data = {
```

Comment:
pass start/end time as arguments - or use a placeholder to be replaced after report generation is completed